cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -329,6 +340,12 @@ void decode_page_headers(hostdevice_vector<gpu::ColumnChunkDesc>& chunks,
   chunks.host_to_device(stream);
   gpu::DecodePageHeaders(chunks.device_ptr(), chunks.size(), stream);
   pages.device_to_host(stream, true);
+
+  // validate page encodings
+  CUDF_EXPECTS(thrust::all_of(
```

Comment:
`thrust::all_of()` continues to have performance challenges: https://github.com/NVIDIA/thrust/issues/1016

Could we consider switching this to `thrust::reduce()` instead?

Reply:
sure

Reply:
Recommend using `thrust::count_if() > 0`.